addon/-legacy-private/system/relationships/state/belongs-to.js

```diff
@@ -156,11 +169,50 @@ export default class BelongsToRelationship extends Relationship {
   }
 
   findRecord() {
-    if (this.inverseInternalModel) {
-      return this.store._findByInternalModel(this.inverseInternalModel);
+    let promise;
+    let { inverseInternalModel, shouldForceReload } = this;
+
+    if (inverseInternalModel) {
+      if (shouldForceReload && inverseInternalModel.hasRecord) {
+        // reload record, if it is already loaded
+        promise = inverseInternalModel.getRecord().reload();
+      } else {
+        promise = this.store._findByInternalModel(inverseInternalModel);
+      }
+      promise = promise.then(
+        () => {
+          return this._handleCompletedFind();
+        },
+        e => {
+          return this._handleCompletedFind(e);
```

Comment:
splitting this into two methods might be much nicer

Reply:
👍